src/main/java/gregtech/common/covers/filter/BaseFilterContainer.java

```diff
@@ -0,0 +1,170 @@
+    private int transferSize;
+    protected final FilterSlotHandler filterInventory;
+
+    public BaseFilterContainer(IDirtyNotifiable dirtyNotifiable) {
+        this.dirtyNotifiable = dirtyNotifiable;
+        filterInventory = new FilterSlotHandler();
+    }
+
+    public ItemStackHandler getFilterInventory() {
+        return filterInventory;
+    }
+
+    public int getMaxTransferSize() {
+        return hasFilter() ? currentFilter.getMaxTransferSize() : this.maxTransferSize;
+    }
+
+    public void setMaxTransferSize(int maxTransferSize) {
+        this.maxTransferSize = MathHelper.clamp(maxTransferSize, 1, Integer.MAX_VALUE);
+        this.transferSize = MathHelper.clamp(this.transferSize, 1, this.maxTransferSize);
+        if (hasFilter()) currentFilter.setMaxTransferSize(this.maxTransferSize);
+    }
+
+    public boolean hasFilter() {
+        return currentFilter != null;
+    }
+
+    public T getFilter() {
+        return currentFilter;
+    }
+
+    public final void setFilter(T newFilter) {
+        this.currentFilter = newFilter;
+        if (currentFilter != null) {
+            currentFilter.setDirtyNotifiable(dirtyNotifiable);
+        }
+        if (onFilterInstanceChange != null) {
+            this.onFilterInstanceChange.run();
+        }
+    }
+
+    public void onFilterInstanceChange() {
+        dirtyNotifiable.markAsDirty();
+    }
+
+    public void setOnFilterInstanceChange(Runnable onFilterInstanceChange) {
+        this.onFilterInstanceChange = onFilterInstanceChange;
+    }
+
+    public boolean showGlobalTransferLimitSlider() {
+        return getMaxTransferSize() > 0 &&
+                (isBlacklistFilter() || !hasFilter() || currentFilter.showGlobalTransferLimitSlider());
+    }
+
+    public int getTransferLimit(int slotIndex) {
+        if (isBlacklistFilter() || currentFilter == null) {
+            return getTransferSize();
+        }
+        return currentFilter.getTransferLimit(slotIndex, getTransferSize());
+    }
+
+    public boolean test(R toTest) {
+        return !hasFilter() || getFilter().test(toTest);
+    }
+
+    public MatchResult<R> match(R toMatch) {
+        if (!hasFilter())
+            return new MatchResult<>(true, toMatch, -1);
+
+        return getFilter().match(toMatch);
+    }
+
+    public int getTransferLimit(R stack) {
+        if (isBlacklistFilter() || currentFilter == null) {
+            return getTransferSize();
+        }
+        return currentFilter.getTransferLimit(stack, getTransferSize());
+    }
+
+    protected abstract void onFilterSlotChange(boolean notify);
+
+    public int getTransferSize() {
+        if (!showGlobalTransferLimitSlider()) {
+            return getMaxTransferSize();
+        }
+        return this.transferSize;
+    }
+
+    public void setTransferSize(int transferSize) {
+        this.transferSize = MathHelper.clamp(transferSize, 1, getMaxTransferSize());
+        onFilterInstanceChange();
+    }
+
+    @Deprecated
```

Comment:
Why is this new method deprecated?

Reply:
this method was originally from `ItemFilterContainer`. it doesn't seem to be used anywhere anymore, so it could be removed